File: 1000/1715B/main.cpp
#include<bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while(t--) 
    {
        long long n, k, b, s;
        cin >> n >> k >> b >> s;

        // Check boundary conditions:
        // Minimum possible sum: b * k
        // Maximum possible sum: b * k + n * (k - 1)
        if(s < b * k || s > (b * k + n * (k - 1))) 
        {
            cout << -1 << "\n";
            continue;
        }

        vector<long long> arr(n, 0);

        // Assign the base value needed to achieve beauty 'b'
        arr[0] = b * k;
        long long remaining_sum = s - b * k;

        // Distribute the remaining sum (up to k - 1 per element)
        for(int i = 0; i < n && remaining_sum > 0; i++) 
        {
            long long add = min(remaining_sum, k - 1);
            arr[i] += add;
            remaining_sum -= add;
        }

        // Print the array
        for(int i = 0; i < n; i++)
            cout << arr[i] << (i == n - 1 ? "" : " ");
        cout << "\n";
    }

    return 0;
}

File: 1000/1715B/main.java
import java.util.*;
public class main
{
	public static void main(String args[]) 
	{
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
 
		while(t-- > 0)
		{
			long n = sc.nextLong();
			long k = sc.nextLong();
			long b = sc.nextLong();
			long s = sc.nextLong();
 
			if(s < b*k || s > (k*b+(k-1)*n))
			{
			    System.out.println(-1);
			    continue;
			}
 
			long sum = 0;
		    long arr[] = new long[(int)n];
 
		    sum = s-b*k;
		    arr[0] = b*k;
			
			for(int i=0;i<n && sum>0;i++)
			{
				long add = Math.min(sum, k-1);
				arr[i] += add;
				sum -= add;
			}
 
			for(int i=0;i<n;i++)
			    System.out.print(arr[i] + " ");
			System.out.println();
		}
	}
}